openedx_lti_tool_plugin/tests/test_views.py

```diff
@@ -383,7 +377,75 @@ def test_get_jwks_mock(self, get_jwks_mock: MagicMock):
         self.assertEqual(response.status_code, 200)
         self.assertJSONEqual(response.content, get_jwks_mock())
 
-    @override_settings(OLTTP_ENABLE_LTI_TOOL=False)
+    @override_settings(OLTITP_ENABLE_LTI_TOOL=False)
+    def test_with_lti_disabled(self):
+        """Test raise 404 response when plugin is disabled."""
+        with self.assertRaises(Http404):
+            self.view_class.as_view()(self.factory.get(self.url))
+
+
+class TestLtiXBlockView(FactoryMixin, TestCase):
+    """Test LTI XBlock view."""
+
+    def setUp(self):
+        """Test fixtures setup."""
+        super().setUp()
+        self.url = reverse('lti-xblock', args=[UNIT_KEY])
+        self.view_class = LtiXBlockView
+
+    @patch('openedx_lti_tool_plugin.views.render_xblock')
+    def test_get_with_usage_key_string(self, render_xblock_mock: MagicMock):
+        """Test GET request with usage key string.
+
+        Args:
+            render_xblock_mock: Mocked render_xblock function.
+        """
+        request = self.factory.get(self.url)
+        self.view_class.as_view()(request, UNIT_KEY)
+
+        self.assertEqual(request.META['HTTP_REFERER'], settings.LEARNING_MICROFRONTEND_URL)
+        render_xblock_mock.assert_called_once_with(request, UNIT_KEY, check_if_enrolled=False)
+
+    @override_settings(OLTITP_ENABLE_LTI_TOOL=False)
+    def test_with_lti_disabled(self):
+        """Test raise 404 response when plugin is disabled."""
+        with self.assertRaises(Http404):
+            self.view_class.as_view()(self.factory.get(self.url))
+
+
+class TestLtiCoursewareView(FactoryMixin, TestCase):
```

Comment:
Shouldn't it also return 404 in case LTI is not active? @kuipumu

Reply:
@kuipumu There is already a test_with_lti_disabled on TestLtiCoursewareView to test this scenario